internal/storage/bucket_handle.go

```diff
@@ -311,3 +311,50 @@ func (b *bucketHandle) UpdateObject(ctx context.Context, req *gcs.UpdateObjectRe
 
 	return
 }
+
+func (b *bucketHandle) ComposeObjects(ctx context.Context, req *gcs.ComposeObjectsRequest) (o *gcs.Object, err error) {
+	dstObj := b.bucket.Object(req.DstName)
+
+	if req.DstGenerationPrecondition != nil {
+		dstObj = dstObj.If(storage.Conditions{GenerationMatch: *req.DstGenerationPrecondition})
+	}
+	if req.DstMetaGenerationPrecondition != nil {
+		dstObj = dstObj.If(storage.Conditions{MetagenerationMatch: *req.DstMetaGenerationPrecondition})
+	}
+	if req.DstGenerationPrecondition != nil && req.DstMetaGenerationPrecondition != nil {
```

Comment:
isn't this if condition same as above two if conditions?

Reply:
No, It is like 
1. if b!=nil
    a=b
2. if c!=nil
    a=c
3. if b!=nil & c!=nil
    a={b,c}

so if the second condition went right then a will not have b , that's why I added third condition

Reply:
I dont see the same logic in create object, so is it broken
https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/storage/bucket_handle.go#L115

Reply:
For createObject the object does not exist so GenerationPrecondition remains zero , that's why it didn't go in any If block, I will fix it for these scenarios

Reply:
okay,.
Also organize if-else more clearly.
if b!=nil & c!=nil

else if b!=nil

else if c!=nil

Reply:
fixed